packages/world/src/World.sol

```diff
@@ -200,7 +188,21 @@ contract World is Store, IWorldCore, IErrors {
    */
   function registerSchema(uint256 tableId, Schema valueSchema, Schema keySchema) public virtual {
     bytes32 tableSelector = ResourceSelector.from(tableId);
-    _this.registerTable(tableSelector.getNamespace(), tableSelector.getFile(), valueSchema, keySchema);
+    (address systemAddress, ) = Systems.get(ResourceSelector.from(ROOT_NAMESPACE, REGISTRATION_SYSTEM_NAME));
+
+    // We can't call IWorld(this).registerSchema directly because it would lose the msg.sender
```

Comment:
ooc what do you mean "lose" the `msg.sender`?

Reply:
`msg.sender` will be the address of the World contract, because calls to `IWorld(this)` are handled like external calls. We could use `delegatecall` to `this`, but that would defeat the purpose of having the `IWorld` interface, in which case using `Call.withSender` directly is cheaper. Will update the comment to make it more clear.